### PR TITLE
Pcse2: irrigation challenge

### PR DESCRIPTION
**Pull request by teytaud**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

The previous crop management tool was cool, but rather preliminary: this was about hyperparameter optimization. The present challenge is about optimizing the irrigation of crops,
making the problem higher-dimensional and more impactful.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](https://facebookresearch.github.io/nevergrad/contributing.html) document and completed the CLA (see [**CLA**](https://facebookresearch.github.io/nevergrad/contributing.html#contributor-license-agreement-cla)).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/irrigation/data/soil/ec3.soil [hunk 1]
```diff
@@ -0,0 +1,55 @@
+** $Id: ec3.new 1.2 1997/09/18 17:33:54 LEM release $
```
Comment (jrapin) [inline nevergrad/functions/irrigation/data/soil/ec3.soil:1]: what's the license with the data files? This is dangereous

Comment (teytaud) [reply nevergrad/functions/irrigation/data/soil/ec3.soil:1]: https://joinup.ec.europa.eu/collection/eupl/matrix-eupl-compatible-open-source-licences suggests that it is ok: we have incorporation rights.


That said you know license stuff 1000x beter than me :)

Comment (jrapin) [reply nevergrad/functions/irrigation/data/soil/ec3.soil:1]: it's really unclear to me, from what I understand of this matrix it means that it's possible to include MIT licenses code into EUPL, but we're doing the reverse, and I can't find the information

Comment (teytaud) [reply nevergrad/functions/irrigation/data/soil/ec3.soil]: Workaround: download from www.

**Pull request merged**